### PR TITLE
migrate from proposal instead of role

### PR DESCRIPTION
**Pull request by MaximilianMeister**
SchemaMigration.migrate_proposal_from_json needs a proposal hash and
cannot deal with any Chef::Role as the attributes is called
default_attributes (and could be different) and there is no deployment
information either

### CODE REVIEW

**Comment by rhafer** [comment]
This will break ```upgrade_knife_files_1_9``` I guess?

**Comment by MaximilianMeister** [comment]
@rhafer arrg, good catch. I guess that would have gone unnoticed if we don't backport the patch... but I will adapt it, thanks

File: crowbar_framework/lib/crowbar/upgrade.rb [hunk 1]
```diff
@@ -115,8 +115,20 @@ def upgrade_crowbar_files_1_9
...
-      json = JSON.load(json_path.read)
+    def create_network_json(json_path, options = {})
+      from_role = options.fetch(:from_role, false)
+      json = if from_role
+        # get network proposal data from database.yml
+        proposals = YAML.load_file(
+          @data.join("crowbar", "database.yml")
+        )
+        network_proposal = proposals["proposals"]["records"].detect do |p|
+          p[1] == "network" && p[2] == "default"
```
Comment (rhafer) [inline crowbar_framework/lib/crowbar/upgrade.rb]: Are you sure that the columns are always in the same order in that file? Or would you need to use ```proposals["proposals"]["columns"]``` to find the correct index for the array element?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/upgrade.rb]: It is always the same order, like defined in the `schema.rb` database schema. and I just looked at 2 different database.yml files. so I guess it is not necessary to map it to the columns.

**Comment by rhafer** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Pull request merged**